src/plugins/intel_npu/src/compiler/include/graph_transformations.hpp

```diff
@@ -14,9 +17,46 @@
  */
 namespace intel_npu::driverCompilerAdapter {
 
-/**
- * @brief Serialize OpenVINO model to IR
- */
-IR serializeToIR(const std::shared_ptr<const ov::Model>& model, uint32_t supportedVersionByCompiler = 7);
+class IR {
+public:
+    IR(const std::shared_ptr<const ov::Model>& origModel,
+       uint32_t supportedVersionByCompiler = 11,
+       bool largeModel = false);
+
+    std::istream& getXml() {
```

Comment:
Why we need to keep two implementations? Could we just use new streambuf?
Alternatively a template could be a way to parametrize without runtime logic

Reply:
For large model like llama and small model like googlnet-v1, the performance of streams are different. I just do a rough check on ubuntu and find follow results:  
For googlenet, 
fps:  stringstream > customStreamBuf > filestream
For llama,
fps: filestream > customStreamBuf > stringstream

And peak value of memory consumption for llama:
fstream(It writes to disk) < customSteam < stringstream

Need to do more tests when Windows driver is ready.  Before that, would keep both stream since most of our models are still smaller than 2G.

Reply:
How about we always stick to stringstream for XML and custom buffer for weights?
I guess for small model the XML handling may be dominant, due to all character fiddling.
and for llama weight processing is likely dominant due to size